internal/config/app_config_sort.go

```diff
@@ -0,0 +1,182 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Simple wrapper around the hclwrite call to turn a traversal into a string.
+func renderTraversal(t hcl.Traversal) (string, error) {
+	tokens := hclwrite.TokensForTraversal(t)
+	var buf bytes.Buffer
+
+	_, err := tokens.WriteTo(&buf)
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+// Captures the information about a variable calculated during sort to be used
+// without recalculation by the caller.
+type analyzedPair struct {
+	Pair     hcl.KeyValuePair
+	Name     string
+	Internal bool
+	Refs     []string
+}
+
+type VariableLoopError struct {
```

Comment:
Comment on this.

Second, since its exported, I would consider putting it towards the top of the file so that people browsing the file may see it sooner but thats purely a matter of personal taste.

Reply:
Done!